seastar/src/api_safety.rs

```diff
@@ -0,0 +1,82 @@
+#[cxx::bridge(namespace = "seastar")]
+mod ffi {
+    unsafe extern "C++" {
+        include!("seastar/core/reactor.hh");
+
+        /// Checks whether the current thread is within a Seastar runtime.
+        fn engine_is_ready() -> bool;
+    }
+}
+
+pub use ffi::engine_is_ready;
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::AppTemplate;
+    use std::thread;
+
+    #[test]
+    fn test_engine_is_ready_only_for_thread_in_runtime() {
+        assert!(!engine_is_ready());
+
+        thread::spawn(|| {
+            assert!(!engine_is_ready());
+            let _guard = crate::acquire_guard_for_seastar_test();
+            let mut app = AppTemplate::default();
+            let fut = async {
+                assert!(engine_is_ready());
+                Ok(())
+            };
+            app.run_void(std::env::args(), fut);
+            // Currently, Seastar does not perform a cleanup of its
+            // thread locals, which unfortunately means
+            // the following assertion  will pass.
+            assert!(engine_is_ready());
```

Comment:
If this problem is ever fixed is seastar, we will have to make some changes, like removing https://github.com/zpp-2022-rust-seastar/seastar-rs/blob/main/seastar/src/seastar_test_guard.rs
Please mention in the comment, that in a case this test fails, we need to look trough project's code and modify our assumptions accordingly - not just remove this line in the test,

Reply:
Sure, that's reasonable.